src/open_prime_rando/echoes/specific_area_patches/required_fixes.py

```diff
@@ -0,0 +1,240 @@
+import logging
+from collections.abc import Iterable
+
+from retro_data_structures.enums.echoes import Message, State
+from retro_data_structures.formats.mrea import Area
+from retro_data_structures.formats.script_object import InstanceIdRef
+from retro_data_structures.properties.echoes.archetypes.EditorProperties import EditorProperties
+from retro_data_structures.properties.echoes.archetypes.LayerSwitch import LayerSwitch
+from retro_data_structures.properties.echoes.archetypes.Transform import Transform
+from retro_data_structures.properties.echoes.core.Vector import Vector
+from retro_data_structures.properties.echoes.objects import (
+    HUDMemo,
+    Pickup,
+    Relay,
+    ScriptLayerController,
+    SpawnPoint,
+    Switch,
+    Timer,
+    Trigger,
+)
+
+from open_prime_rando.echoes.asset_ids import agon_wastes, sanctuary_fortress, temple_grounds, torvus_bog
+from open_prime_rando.echoes.asset_ids.world import (
+    AGON_WASTES_MLVL,
+    SANCTUARY_FORTRESS_MLVL,
+    TEMPLE_GROUNDS_MLVL,
+    TORVUS_BOG_MLVL,
+)
+from open_prime_rando.patcher_editor import PatcherEditor
+
+LOG = logging.getLogger("echoes_patcher")
+
+
+def required_fixes(editor: PatcherEditor):
```

Comment:
The scan for activating the cannon in Sacred Bridge is deactivated after lowering the gate (and I think reloading the room). Keeping the scan always active will prevent players from softlocking down there if they have no items to climb back up and the cannon wasn't activated prior.